packages/tpetra/core/inout/MatrixMarket_Tpetra.hpp

```diff
@@ -3673,19 +3673,17 @@ namespace Tpetra {
                   << prvRow << ", at curPos = " << curPos << ".  Please report "
                   "this bug to the Tpetra developers.");
                 if (curRow > prvRow) {
-                  for (global_ordinal_type r = prvRow+1; r <= curRow; ++r) {
-                    rowPtr[r] = curPos;
-                  }
```

Comment:
From #11704 
> This skips setting all row-pointer array values between the row with the last non-zero, and the final row.

@cwpearson Is there a test that will expose this error?

Reply:
I don't think so, let me see if I can remedy that

Reply:
I have added one